packages/react-components/src/components/abstract-xy-output-component.tsx

```diff
@@ -381,12 +385,28 @@ export abstract class AbstractXYOutputComponent<P extends AbstractOutputProps, S
         );
     }
 
-    private async updateXY(): Promise<void> {
-        if (document.getElementById(this.getOutputComponentDomId() + 'handleSpinner')) {
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            document.getElementById(this.getOutputComponentDomId() + 'handleSpinner')!.style.visibility = 'visible';
+    /**
+     * Spread calls to fetchXY().
+     * If fetchTree() calls this.updateXY() directly, it can cause too many calls to
+     * this.props.tspClient.fetchXY(), all very close to each other, which can
+     * lead to errors.
+     * updateXY() is called repeatedly so that the user can see that data is being
+     * loaded, which would no tbe possible if using debounce.
+     */
+    private async recursiveFetch(): Promise<void> {
+        if (!this.isFetching) {
+            this.isFetching = true;
+            const timeout = 3000;
+            while (this.props.experiment?.indexingStatus === ResponseStatus.RUNNING ||
```

Comment:
indexingStatus is independent from the analysis execution. You only need to check the outputStatus

Reply:
Checks for indexing status removed.